riscv/interactive.cc

```diff
@@ -151,14 +154,27 @@ void sim_t::interactive()
 
   while (!done())
   {
-    std::string s;
 #ifdef HAVE_BOOST_ASIO
     streambuf bout; // socket output
-    s = rin(&bout); // get command string from socket or terminal
+#endif
+    std::string s;
+    char cmd_str[MAX_CMD_STR]; // only used for following fscanf
+    // first get commands from file, if cmd_file has been set
+    if (cmd_file && fscanf(cmd_file,"%" STR(MAX_CMD_STR) "[^\n]\n", cmd_str)==1) {
+                                  // up to MAX_CMD_STR characters before \n, skipping \n
+       s = cmd_str;
+       // while we get input from file, output goes to stderr
+       sout.rdbuf(cerr.rdbuf());
+    } else {
+       // when there are no commands left from file or if there was no file from the beginning
+       cmd_file = NULL; // mark file pointer as being not valid, so any method can test this easily
```

Comment:
was this resolved?

Reply:
Yes,
line 163 of riscv/interactive.cc was changed from
`if (cmd_file && ...`
to
`if (cmd_file && !feof(cmd_file) && ...`
so that reaching EOF goes to the `else` where 'cmd_file = NULL` gets executed.